### PR TITLE
Collect user feedback

### PR DESCRIPTION
**Pull request by swalsh15**
Fixed requirements and imports so firebase works when deployed on heroku. Added changes so firebase only is initiated once and entries that don't compile or are empty ie "" are not stored.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -38,14 +38,19 @@ def main(input_code, input_lang, output_lang):
...
-    # don't return error without sending to database
-    output_code = ""
+
     if (type(gast) == str) :
-        # return error if gast not built
-        output_code = "Error: did not compile"
-    else:
-        output_code = gtc.gast_to_code(gast, output_lang)
+        # return error if gast not built - dont store in database
+        return "Error: did not compile"
```
Comment (g-harel) [inline main.py:44]: I'd prefer if there was a stronger link between this error response and the condition being checked in the data_collection code. Maybe a constant string or error code?

File: data_collection.py [hunk 2]
```diff
@@ -13,8 +13,10 @@ def store_query(input_code, output_code, input_lang, output_lang):
         print("Please set GOOGLE_APPLICATION_CREDENTIALS environmental variable. See readme for more info")
         return
     
-    cred = credentials.Certificate(cred_object)
-    firebase_admin.initialize_app(cred)
+    # only initialize firebase once
+    if not firebase_admin._apps:
```
Comment (augerm) [inline data_collection.py:17]: I would wrap this function in a `DataService` [singleton](https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html) class. That way it is only initialized once. Happy to hop on a call and walk through this!

File: tools/add_environ_var.py [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
+import subprocess
 
 try:
     with open("../private_keys/firebase-access-key.json") as json_file:
         data = json.load(json_file)
+        env = {
+            **os.environ,
+            "GOOGLE_APPLICATION_CREDENTIALS": json.dumps(data),
+        }
+        subprocess.Popen('/bin/bash', env=env).wait()
```
[the remark this answers is not in the document]
Comment (swalsh15) [reply tools/add_environ_var.py:12]: This exports the variable and allows it to be used in subprocesses. The old code created a local variable that could only be used in the file it was created in

File: main.py [hunk 4]
```diff
@@ -38,14 +38,19 @@ def main(input_code, input_lang, output_lang):
...
    
-    # don't return error without sending to database
-    output_code = ""
+
     if (type(gast) == str) :
-        # return error if gast not built
-        output_code = "Error: did not compile"
-    else:
-        output_code = gtc.gast_to_code(gast, output_lang)
+        # return error if gast not built - dont store in database
```
Comment (jackdavidweber) [inline main.py:43]: why not store this in the database? Wouldn't it be a useful thing to know if people aren't able to get code to compile?

File: main.py [hunk 5]
```diff
@@ -38,14 +38,19 @@ def main(input_code, input_lang, output_lang):
...
+        return "Error: did not compile"
+   
+    output_code = gtc.gast_to_code(gast, output_lang)
     
-    #data_collection.store_query(input_code, output_code, input_lang, output_lang)
+    # don't store empty strings
```
Comment (jackdavidweber) [inline main.py:48]: Maybe elaborate on comment

File: main.py (hunk 4 above)
Comment (swalsh15) [reply main.py:43]: Good idea - going to store and categorize whether successful, didn't compile, or feature not supported

File: main.py (hunk 5 above)
Comment (swalsh15) [reply main.py:48]: Got it will add to it

**Pull request merged**